src/main.rs

```diff
@@ -170,18 +192,52 @@ fn sync_list(path: &str, pattern: &str, list_id: &str) -> Result<(), Box<std::er
     let client = letterboxd::Client::new(&core.handle(), key, secret);
     let do_auth = client.auth(&username, &password);
     let token = core.run(do_auth)?;
-    print!("Got token: {:?}", token);
+    println!("Got token: {:?}", token);
 
     let files = list_files(path)?;
 
-    // Fetch ids for films on path.
+    // Collect all movie names
     let re = Regex::new(pattern)?;
     let movie_names = files.filter_map(|file_name| extract_movie(&re, file_name.as_str()));
-    let requests = movie_names.map(|movie| search_movie(&client, movie));
-    let film_ids = future::join_all(requests).map(|responses| -> Vec<_> {
-        responses
+
+    // Load ids either from cache file or make a request to get an id
+    let film_ids_cache = load_ids_list_from_cache()?;
+    let film_ids_req = movie_names.map(|movie| -> Box<
+        Future<
+            Item = (String, String),
+            Error = letterboxd::Error,
+        >,
+    > {
+        let id = film_ids_cache.get(&movie);
+        // TODO: Try to remove Box here
+        if let Some(id) = id {
+            Box::new(future::ok((movie, id.clone())))
+        } else {
+            Box::new(search_movie(&client, movie.clone()).and_then(
+                move |mut resp| {
```

Comment:
Macht es Sinn diesen Teil in einer Hilfsmethod zu schieben? Würde das leserlicher sein?

Reply:
done